### PR TITLE
Development branch

### PR DESCRIPTION
**Pull request by gisachris**
## IN This PR We present the SpaceX travelers application:🌐 
**This application has undergone its development 👨‍💻 phase and completed successfully**

## Requirements
- All requirements can be found [here](https://github.com/microverseinc/curriculum-react-redux/blob/main/group-project/project_space_travelers_hub.md)

## ACHIEVEMENTS ⛰️
- Application has a fully built and well-designed UI🎨 and UX⚙️ system ready to fulfill all these [requirements](https://github.com/microverseinc/curriculum-react-redux/blob/main/group-project/project_space_travelers_hub.md)

## Conclusion
**We would love to hear 👂  from you Our CR in terms of what can be better implemented 👨‍💻  for the site thank you! And we look forward to hearing from you soon.**  🙇

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import './styles/general.css';
+import './styles/profile.css';
+import RocketsDisplay from './components/rocketsDisplay';
+
+function App() {
+  const dispatch = useDispatch();
+  useEffect(() => {
+    dispatch(fetchData());
+    dispatch(fetchMissions());
+  }, []);
```
Comment (tugsem) [inline src/App.js]: - [ ] Rockets is the default view, so its data would be fetched when the application starts. However, the missions data should only be fetched (once) when a user navigates to the Missions section. Kindly fix this.

File: src/redux/features/rocketView.jsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+          )}
+
+          {obj.reserved && (
+          <button type="submit" className="cancelRocket" onClick={() => handlecancelation(obj.id)}>
+            Cancel Reservation
+          </button>
+          )}
+
+        </section>
+      </div>
```
Comment (tugsem) [inline src/redux/features/rocketView.jsx]: - [ ] For the best JavaScript practices and the best readability, this part can be imported from a rocket component. Also, this file shouldn't be in the redux directory. Please move it to the components folder like the MissionList.js and Mission.js files: 
image

File: src/components/DisplayMissions.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+import '../styles/profile.css';
+
+function DisplayMissions() {
+  const missions = useSelector((store) => store.missions.missions);
+  const joinedMissions = missions.filter((mission) => mission.joined);
+  return (
+    <div className="missions">
+      <h2>Missions</h2>
+      {joinedMissions.length === 0 ? (
+        <p>No missions have been added yet!</p>
```
Comment (tugsem) [inline src/components/DisplayMissions.js:13]: - Kudos for adding a placeholder message here! 👏

File: src/components/Profile.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  const bookedRockets = rocketData.filter((rockets) => rockets.reserved === true);
+  return (
+    <div className="profile">
+      <div className="missions">
+        <DisplayMissions />
+      </div>
+      <div className="rockets">
+        <h2 className="rocketsHeader">Rockets</h2>
+        {bookedRockets.length === 0 && (
+          <p>No rockets have been reserved yet!</p>
```
Comment (tugsem) [inline src/components/Profile.js]: - Kudos for adding a placeholder message here! 👏

**Review by tugsem — CHANGES_REQUESTED** [review]
##  STATUS: REQUIRED CHANGES ♻️
Hello @gisachris  and @husush23 👋 , 

Good job so far! 👏 
There are some issues that you still need to work on to go to the next project but you are almost there!

Alt Text
## Highlights 🚀 
✔️Redux store, API requests are created correctly
✔️readme & documentation are professional
✔️ the app works correctly
✔️ Integration tests are successful
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Profile.js (hunk 4 above)
Comment (gisachris) [reply src/components/Profile.js]: @tugsem, Thank you for the great review 👏. We will get right on it and fix⚙️ these issues.

**Comment by gisachris** [comment]
## Implemenentations of the Required Changes ♻️:
Below are the implementations from the required Changes♻️

## Requirements
- Move missions `fetch` 📞 request to the `/missions` route 🌉 as per CodeReviewer 's requirements [check here](https://github.com/gisachris/react-group-project/pull/43#discussion_r1246606166)
- Move the `rocketView.jsx` from the redux ⚛️ logic directory `redux` into the `components` Directory.

## Achievements
- Move missions `fetch` 📞 request to the `/missions` route 🌉 as per CodeReviewer 's requirements [check here](https://github.com/gisachris/react-group-project/pull/43#discussion_r1246606166) ✅ 
- Move the `rocketView.jsx` from the redux ⚛️ logic directory `redux` into the `components` Directory. ✅

## N.B
As per the previous code Reviewer's request [check here](https://github.com/gisachris/react-group-project/pull/43#discussion_r1246606166), Changing the missions fetch 📞 command from the `appDirectory` into the `missions` route directory means once a user redirects to the `/missions` route again the previous data will be lost. `forExample`: The data displayed in the `profile` view 

## Conclusion
**We would love to hear 👂  from you Our CR in terms of what can be better implemented 👨‍💻  for the site thank you! And we look forward to hearing from you soon.**  🙇

File: src/components/rocketView.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+          {obj.reserved && (
+          <button type="submit" className="cancelRocket" onClick={() => handlecancelation(obj.id)}>
+            Cancel Reservation
+          </button>
+          )}
+
+        </section>
+      </div>
+    ))
+  );
```
Comment (danmainah) [inline src/components/rocketView.js]: - [ ] @gisachris and @husush23 great work implementing the changes requested however you forgot to implement the feature requested adding an additional component for easy readability. You can create a `Rocket.js` file that handles the importation of rocket data from redux them passes it to this rocketView component via props like you did for thew Missions and Mission components

File: src/components/Missions.js [hunk 6]
```diff
@@ -0,0 +1,41 @@
+/*eslint-disable*/
```
Comment (danmainah) [inline src/components/Missions.js]: - [ ] @team great work ensuring the missions are fetched only when the user accesses the missions page, however when implementing it you disabled the linter check which is not a good practise, linter checks help us to write better code, kindly consider enabling linter check for this file

File: src/__tests__/navComponent.test.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
```
Comment (danmainah) [inline src/__tests__/navComponent.test.js:1]: - [x] [optional] Great work on your tests, however, I would recommend you add more tests for all other components and pure methods in redux directory

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @gisachris and @husush23 , 

Great work on making the changes requested by a previous reviewer 👏  👏 👏
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
✔️  Implemented changes requested ensuring the missions' data is only fetched when the user accesses the missions page
✔️  Implemented changes requested ensuring the Directory structure for file is better organized.


 # Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by gisachris** [comment]
## Implemenentations of the Required Changes ♻️:
Below are the implementations from the required Changes♻️

## Requirements
- Remove `linter disabling` ❌ as a good practice.
- Create a new component 🏡 for a single rocket 🚀 in `rocket.js` to implement better readability.
- Add a test 🧪 for the created component. 

## Achievements
- Remove `linter disabling` ❌ as a good practice.✅
- Create a new component 🏡 for a single rocket 🚀 in `rocket.js` to implement better readability.✅
- Add a test 🧪 for the created component. ✅

## Conclusion
**We would love to hear 👂  from you Our CR in terms of what can be better implemented 👨‍💻  for the site thank you! And we look forward to hearing from you soon.**  🙇

**Review by DeliceLydia — APPROVED** [review]
## Hi Team, 

In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 


### Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Comment by gisachris** [comment]
@DeliceLydia, t=Thank you for the great review 👏

**Pull request merged**